python/ngen_cal/src/ngen/cal/_plugin_system.py

```diff
@@ -27,3 +27,17 @@ def setup_plugin_manager(plugins: list[Callable | ModuleType]) -> PluginManager:
             assert_never(plugin)
 
     return pm
+
+def setup_scoped_plugin_manager(spec: type, plugins: list[Callable | ModuleType]) -> PluginManager:
```

Comment:
```suggestion
def setup_namespaced_plugin_manager(spec: type, plugins: list[Callable]) -> PluginManager:
```

I think the type hints here can be cleaned up slightly just emphasize the invariants at play.

Reply:
changed in a4710a941b0869d1bf1c2d2bfd920b93b09646fc